src/Microsoft.ML.AutoML/Experiment/Experiment.cs

```diff
@@ -196,6 +196,22 @@ public IList<TRunDetail> Execute()
                         "was reached, and the running MLContext was stopped. Details: {0}", e.Message);
                     return iterationResults;
                 }
+                catch (AggregateException e)
+                {
+                    // This exception is thrown when the IHost/MLContext of the trainer is canceled due to
+                    // reaching maximum experiment time. Simply catch this exception and return finished
+                    // iteration results. For some trainers, like FastTree, because training is done in parallel
+                    // in can throw multiple OperationCancelledExceptions. This causes them to be returned as an
+                    // AggregateException and misses the first catch block. This is to handle that case.
+                    if (e.InnerExceptions.All( exception => exception is OperationCanceledException))
+                    {
+                        _logger.Warning("OperationCanceledException has been caught after maximum experiment time" +
+                        "was reached, and the running MLContext was stopped. Details: {0}", e.Message);
+                        return iterationResults;
+                    }
+
+                    throw;
+                }
             } while (_history.Count < _experimentSettings.MaxModels &&
                     !_experimentSettings.CancellationToken.IsCancellationRequested &&
```

Comment:
> I was not sure how to really add tests for this, It was brought up by a user and you can repro it locally though its not consistent. If any of you have ideas for good tests let me know.

To add a unit test focusing on `OperationCanceledException`, you could set the AutoML setting to use only FastTree:
```c#
var experimentSettings = new RegressionExperimentSettings { MaxExperimentTimeInSeconds = 1 };
experimentSettings.Trainers.Clear();
experimentSettings.Trainers.Add(RegressionTrainer.FastTree);
```

Reply:
I have tried that, but it only throws this exception in 1 specific method and this wasn't a stable repro. Not stable enough to try and have a consistent test for it anyways.

Reply:
Agreed. After discussing, even restricting to only FastTree will not make for a stable unit test.

If the timeout occurs in certain steps of FastTree, `AggregateException` is thrown; while at other steps, `OperationCanceledException` is thrown.

Reply:
Could you force it with a test-specific Trainer, or set of trainers?

Reply:
I tried that. The problem is that there is only 1 specific place that causes this issue when a timeout occurs, and I haven't found a way to get it to repro consistently without modifying the method to always throw there.

Reply:
I was thinking more along the lines of a test-only mock trainer.  Ignore if we don't already have a framework / pattern for this, that'd be more of a future-looking test strategy piece of feedback.